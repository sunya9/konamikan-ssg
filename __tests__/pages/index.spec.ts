import { mount } from '@vue/test-utils'
import { post, postWithFeatureImage } from '@/__tests__/fixtures/postOrPage'
import Index from '@/pages/index.vue'

describe('index page', () => {
  it('normal', () => {
    expect(
      mount(Index, {
        mocks: {
          posts: [post, postWithFeatureImage]
        }
      })
    ).toMatchSnapshot()
  })
})
