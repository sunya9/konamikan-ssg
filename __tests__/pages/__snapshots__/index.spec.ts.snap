// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index page normal 1`] = `
<div>
  <header class="hero is-medium fix-background" style="background-image: url(cover.png);">
    <div class="hero-head">
      <nav class="navbar is-transparent">
        <div class="container">
          <div class="navbar-brand">
            <!----> <a href="#" aria-label="Menu" class="navbar-burger has-text-white"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a></div>
          <div class="navbar-menu is-relative">
            <div class="navbar-end">
              <div class="navbar-item">
                <div class="control has-icons-right"><label for="search-form-input" class="is-sr-only">Search</label> <input id="search-form-input" type="search" placeholder="Search..." autocomplete="off" class="input is-small search"> <span class="icon is-small is-right"><svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"><circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line></svg></span></div>
                <!---->
              </div> <a href="/rss" class="navbar-item has-text-white-ter"><span class="icon"><svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-rss"><path d="M4 11a9 9 0 0 1 9 9"></path><path d="M4 4a16 16 0 0 1 16 16"></path><circle cx="5" cy="19" r="1"></circle></svg></span> <span class="is-sr-only">RSS</span></a>
            </div>
          </div>
        </div>
      </nav>
    </div>
    <div class="hero-body fix-z-index">
      <div class="container has-text-centered">
        <h1 class="title is-1 has-text-white">
          title
        </h1>
        <h2 class="subtitle has-text-white-bis">
          description
        </h2>
      </div>
    </div>
    <!---->
  </header>
  <main class="section">
    <div class="container">
      <div class="columns is-multiline">
        <div class="column is-4">
          <article class="card fix-height">
            <div class="card-image"><a>
                <div aria-labelledby="__test__title" aria-describedby="__test__description" class="image-2by1 has-background-light no-image has-text-secondary">
                  <!---->
                </div>
              </a></div>
            <div class="card-content">
              <div class="content">
                <h3 id="__test__title" class="title is-5"><a>
                    title
                  </a></h3>
                <h4 class="subtitle is-6 has-text-secondary"><small>
                    2020/04/05
                  </small></h4>
                <p id="__test__description">
                  excerpt
                </p>
              </div>
            </div>
          </article>
        </div>
        <div class="column is-4">
          <article class="card fix-height">
            <div class="card-image"><a>
                <div aria-labelledby="__test__title" aria-describedby="__test__description" class="image-2by1"><img src="feature_image.png" alt="" loading="lazy" class="image-2by1-inner"></div>
              </a></div>
            <div class="card-content">
              <div class="content">
                <h3 id="__test__title" class="title is-5"><a>
                    title
                  </a></h3>
                <h4 class="subtitle is-6 has-text-secondary"><small>
                    2020/04/05
                  </small></h4>
                <p id="__test__description">
                  excerpt
                </p>
              </div>
            </div>
          </article>
        </div>
      </div> <a class="button is-fullwidth is-rounded is-outlined is-text">
        Archives
      </a>
    </div>
  </main>
</div>
`;
